### ISSUE TITLE
o-forms: duplicate ID used in demo - ID: DAC_Parsing_02

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#545]
# Issue
This is from page 154 of the [audit report](https://drive.google.com/file/d/17wq6b0CIX41RA4muBFDzUDxE0-qzHBIH/view)

Issue can be found on: <https://www.ft.com/__origami/service/build/v3/demo?component=o-forms@9.0.1&demo=interactive&system_code=origami-repo-data&brand=master>

The date input fields have not been grouped together in a fieldset and legend and have been labelled using aria-labelledby, however a duplicate ID value has been used on both the date inputs and the radio buttons below them.
As a result, screen reading software uses the ‘Date input’ label on both the date inputs and the radio buttons when navigating out of the context of the page and the radio buttons appear as unlabelled when navigating in the context of the page.
This may be confusing for screen reader users who may find difficulty in distinguishing between the two sets of inputs.
This issue may also affect voice activation users who will be unable to reference the radio buttons by their visual label.

HTML which has the duplicate ID and aria-labelledby attributes:

> &lt;div class="o-forms-field o-forms-field--optional" role="group" **aria-labelledby="date-group-title"**&gt;
> &lt;span class="o-forms-title"&gt;
> &lt;span class="o-forms-title__main" **id="date-group-title"**&gt;Date input&lt;/span&gt; &lt;span class="o-forms-title__prompt"&gt;Optional prompt text&lt;/span&gt;
> &lt;/span&gt;
> &lt;span class="o-forms-input o-forms-input--date"&gt;
> &lt;label&gt;
> &lt;input class="o-forms-input__day-part" id="date" type="text" name="date"
> value="" pattern="0[1-9]|[12]\d|3[01]" aria-label="Day (DD)" required=""&gt; &lt;span class="o-forms-input__label" aria-hidden="true"&gt;DD&lt;/span&gt; &lt;/label&gt;
> &lt;label&gt;
> &lt;input class="o-forms-input__month-part" type="text" name="date" value="" pattern="0?[1-9]|1[012]" aria-label="Month (MM)" required=""&gt;
> &lt;span class="o-forms-input__label" aria-hidden="true"&gt;MM&lt;/span&gt; &lt;/label&gt;
> &lt;label&gt;
> &lt;input class="o-forms-input__year-part" type="text" name="date" value=""
> pattern="[0-9]{4}" aria-label="Year (YYYY)" required=""&gt;
> &lt;span class="o-forms-input__label" aria-hidden="true"&gt;YYYY&lt;/span&gt;
> &lt;/label&gt;
> &lt;span class="o-forms-input__error"&gt;Please fill out this field with the required
> format (DD/MM/YYYY)&lt;/span&gt;
>   &lt;/span&gt;
> &lt;/div&gt; 

> &lt;div class="o-forms-field" role="group" **aria-labelledby="date-group-title"**&gt; &lt;span class="o-forms-title" aria-hidden="true"&gt;
> &lt;span class="o-forms-title__main" **id="date-group-title"**&gt;Radio box input&lt;/span&gt; &lt;/span&gt;
> &lt;span class="o-forms-input o-forms-input--radio-box"&gt; &lt;div class="o-forms-input--radio-box__container"&gt;
> &lt;label&gt;
> &lt;input id="radio" type="radio" name="box"&gt;
> &lt;span class="o-forms-input__label" aria-hidden="true"&gt;Yes&lt;/span&gt; &lt;/label&gt;
> &lt;label&gt;
> &lt;input type="radio" name="box" checked=""&gt;
> &lt;span class="o-forms-input__label o-forms-input__label--negative" aria-
> hidden="true"&gt;No&lt;/span&gt; &lt;/label&gt;
>     &lt;/div&gt;
>   &lt;/span&gt;
> &lt;/div&gt;

# Proposed Solution:
Ensure the radio inputs have a unique ID attribute which matches the aria-labelledby attribute for the group. This will enable users of assistive technologies to identify and interact with the radios correctly.

**Comment by JakeChampion** [comment]
This looks to have been completed in https://github.com/Financial-Times/origami/releases/tag/o-forms-v9.2.4
